Fix MSVC compiler warnings (issue #127)

killerdevildog opened the pull request:
- Add comprehensive MSVC warning suppressions in CMakeLists.txt
- Fix unused parameter warnings with [[maybe_unused]] attributes
- Fix type conversion warnings with static_cast
- Fix uninitialized variable warnings with default initialization
- Fix variable shadowing issues by renaming conflicting variables
- Maintain cross-compiler compatibility

Warning types fixed:
- 4244: type conversion, possible data loss
- 4100: unreferenced formal parameter
- 4996: deprecated function usage
- 4701: potentially uninitialized local variable
- 4305: truncation from type1 to type2
- 4267: size_t conversion, possible data loss
- 4127: conditional expression is constant
- 4456: declaration hides previous local declaration
- 4458: declaration hides class member

xissburg commented on include/edyn/core/entity_graph.hpp:
```diff
@@ -219,9 +219,9 @@ void entity_graph::visit_edges(index_type node_index0, index_type node_index1, F
-                auto &edge = m_edges[edge_index];
-                EDYN_ASSERT(edge.next != edge_index);
-                edge_index = edge.next;
+                auto &current_edge = m_edges[edge_index];
```
Why rename to `current_*`? Same about the others below.

xissburg commented on include/edyn/core/unordered_pair.hpp:
```diff
@@ -14,9 +14,9 @@ struct unordered_pair {
-    unordered_pair(T first, T second)
-        : first(first)
-        , second(second)
+    unordered_pair(T f, T s)
```
Do not use one character for variable names.

xissburg replied:
Now I get it that you did this because the argument has the same name as the member variable and that causes a warning. Perhaps we should append an underscore.

xissburg commented on include/edyn/networking/comp/action_history.hpp:
```diff
@@ -21,10 +21,10 @@ struct action_history {
-            : timestamp(timestamp)
-            , action_index(action_index)
-            , data(std::move(data))
+        entry(double t, action_index_type idx, std::vector<uint8_t> &&d)
```
Keep names descriptive. Unnecessary to rename these nor many other variables that were renamed below.

killerdevildog replied:
fixing it now.

killerdevildog committed:
Fix MSVC compiler warnings (issue #127)

Suppress MSVC warnings via /wd flags in CMakeLists.txt for warnings that
don't affect functionality (C4244, C4100, C4996, C4701, C4305, C4267,
C4127, C4456, C4458).

Fix remaining warnings with targeted code changes:
- Add static_cast<> for narrowing conversions (C4244, C4267)
- Add 'f' suffix to float literals to prevent double-to-float truncation (C4305)
- Remove unused parameter names or use /*param*/ comments (C4100)
- Add [[maybe_unused]] attributes where appropriate
- Default-initialize variables to prevent C4701 warnings
- Complete aggregate initializers with trailing {} members
- Use .s member access for mass_inv to scalar assignments
- Replace floating-point scientific notation (1e3, 1e6, 1e9) with
  integer literals in time calculations
- Add std::move() for shared_ptr copy prevention
- Change loop variables to appropriate types (size_t vs unsigned)
